snmplib/snmp_api.c

```diff
@@ -1926,7 +1926,7 @@ snmp_free_session(netsnmp_session * s)
          */
         netsnmp_callback_clear_client_arg(s, 0, 0);
 
-        free((char *) s);
```

Comment:
`free(NULL)` has no effect and clearing a variable that won't be reused is not necessary. Hence, this change of `free()` into `SNMP_FREE()` is not useful. Same comment for the other `SNMP_FREE()` change.